src/lib/remark-plugins/rewrite-tutorial-links/index.ts

```diff
@@ -66,32 +66,40 @@ export const rewriteTutorialLinksPlugin: Plugin = () => {
 }
 
 function handleRewriteTutorialsLink(node: Link | Definition) {
+  node.url = rewriteTutorialsLink(node.url, TUTORIAL_MAP)
+}
+
+export function rewriteTutorialsLink(
+  url: string,
+  tutorialMap: Record<string, string>
+): string {
+  let newUrl = url
   try {
     // return early if non tutorial or collection link
-    if (!learnLink.test(node.url) && !docsLink) {
-      return
+    if (!learnLink.test(newUrl) && !docsLink.test(newUrl)) {
+      return newUrl
     }
 
-    const match: RegExpMatchArray | null = node.url.match(
+    const match: RegExpMatchArray | null = url.match(
       new RegExp(`${learnProductOptions}|cloud`)
     )
     const product = match ? match[0] : null
-    const isExternalLearnLink = node.url.includes('learn.hashicorp.com')
+    const isExternalLearnLink = url.includes('learn.hashicorp.com')
     const isBetaProduct = product
       ? getIsBetaProduct(product as ProductSlug)
       : false
     // Anchor links for the current tutorial shouldn't be rewritten. i.e. #some-heading
-    const isAnchorLink = node.url.startsWith('#')
+    const isAnchorLink = url.startsWith('#')
 
     // if its not a beta product and also not an external link, rewrite
     // external non-beta product links don't need to be rewritten. i.e. learn.hashicorp.com/consul
     if (!isBetaProduct && !isExternalLearnLink && !isAnchorLink) {
       // If its an internal link, rewrite to an external learn link
-      node.url = new URL(node.url, 'https://learn.hashicorp.com/').toString()
+      newUrl = new URL(newUrl, 'https://learn.hashicorp.com/').toString()
     }
 
     if (isBetaProduct) {
-      let nodePath = node.url // the path to be formatted - assumes to be absolute as current Learn impl does
+      let nodePath = newUrl // the path to be formatted - assumes to be absolute as current Learn impl does
```

Comment:
```suggestion
      let nodePath = newUrl // the path to be formatted - assumes to be absolute as current Learn impl does
```
```suggestion
      let nodePath = url // the path to be formatted - assumes to be absolute as current Learn impl does
```
It may just be my brain, but it seems easier to use `url` since `newUrl` could change and I need to read the code to check what `newUrl` may be at this point in the code.